centos2ol.sh

```diff
@@ -303,11 +303,25 @@ case "$os_version" in
         fi
         ;;
     8*)
-        # Workaround for qemu-guest-agent packages installed from virt modules
-        if rpm -q qemu-guest-agent; then
-            dnf module reset -y virt
-            dnf module enable -y virt
-            dnf install -y --allowerasing qemu-guest-agent
+        # There are a few dnf modules that are named after the distribution
+        #  for each steam named 'rhel' or 'rhel8' perform a module reset and install
+        mapfile -t modules_enabled < <(dnf module list --enabled | grep rhel | cut -f1 -d\  )
+        if [[ "${modules_enabled[*]}" ]]; then
+            for module in "${modules_enabled[@]}"; do
+                dnf module reset -y "${module}"
+                case ${module} in
+                container-tools|go-toolset|jmc|llvm-toolset|rust-toolset)
+                    dnf module install -y "${module}":ol8
+                    ;;
+                virt)
+                    dnf module install -y "${module}":ol
+                    ;;
+                *)
+                    echo "Unsure how to transform module ${module}"
```

Comment:
We should either break here or  get user confirmation to continue, instead of just logging the exception. 

I would then rewrite this as `"Cannot switch enabled module \"${module}\" to Oracle Linux. Continue switching (y/N):` with "no" as the default option.

Reply:
At this point in the code the core OS has been switched and we're tidying up the edge cases. We can't easily revert the changes already made so I don't think halting would help. I appreciate it's something that should be strongly highlighted to the user to replace a line that may scroll by.

What do you think of a summary at the end indicating that a module $x couldn't be automatically switched and requires manual intervention?

Reply:
I think we should move this test to before we do anything that changes the system so that users have the option to abort the switch before any damage is done.

Reply:
OK, I'll leave this modify code at the end but I'll add a check at the start to see if there are rhel* streams enabled that we aren't familiar with.

Reply:
Thant seems like a perfectly cromulent solution.

Reply:
Implemented change. I temporarily disabled knowledge of the virt:rhel stream and this is the output:
```
$ sudo bash up.sh 
Checking for required packages...
Checking your distribution...
Checking for yum lock...
Checking for required python packages...
Identifying dnf modules that are enabled
This tool is unable to automatically switch module 'virt' from a CentOS 'rhel' stream to
an Oracle Linux equivalent. Do you want to continue and resolve it manually?
You may want select No to stop and raise an issue on https://github.com/oracle/centos2ol/ or contact <oraclelinux-info_ww_grp@oracle.com> for advice.
1) Yes
2) No
#? 2
Unsure how to switch module 'virt'. Exiting as requested
For assistance, please email <oraclelinux-info_ww_grp@oracle.com>.
```

Reply:
What if there are multiple unknown modules? This format seems very single-module specific.

Also, only provide the GitHub URL, not the email address.